alicloud/resource_alicloud_slb_domain_extension.go

```diff
@@ -0,0 +1,155 @@
+	req.ServerCertificateId = d.Get("server_certificate_id").(string)
+	err := resource.Retry(3*time.Minute, func() *resource.RetryError {
+		raw, err := client.WithSlbClient(func(slbClient *slb.Client) (interface{}, error) {
+			return slbClient.CreateDomainExtension(req)
+		})
+		if err != nil {
+			if IsExceptedErrors(err, []string{DomainExtensionProcessing}) {
+				return resource.RetryableError(WrapErrorf(err, DefaultErrorMsg, "slb_domain_extension", req.GetActionName(), AlibabaCloudSdkGoERROR))
+			}
+			return resource.NonRetryableError(fmt.Errorf("CreateDomainExtension got an error: %#v", err))
+		}
+		resp := raw.(*slb.CreateDomainExtensionResponse)
+		d.SetId(resp.DomainExtensionId)
+		addDebug(req.GetActionName(), raw)
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+	return resourceAliyunSlbDomainExtensionRead(d, meta)
+}
+
+func resourceAliyunSlbDomainExtensionRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	slbService := SlbService{client}
+	id := d.Id()
+	domainExtension, err := slbService.DescribeDomainExtensionAttribute(id)
+	if err != nil {
+		if NotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return WrapError(err)
+	}
+	d.Set("id", domainExtension.DomainExtensionId)
+	d.Set("load_balancer_id", domainExtension.LoadBalancerId)
+	d.Set("domain", domainExtension.Domain)
+	d.Set("server_certificate_id", domainExtension.ServerCertificateId)
+	d.Set("frontend_port", domainExtension.ListenerPort)
+	return nil
+}
+
+func resourceAliyunSlbDomainExtensionUpdate(d *schema.ResourceData, meta interface{}) error {
+	d.Partial(true)
+	if d.HasChange("server_certificate_id") {
+		req := slb.CreateSetDomainExtensionAttributeRequest()
+		req.DomainExtensionId = d.Id()
+		req.ServerCertificateId = d.Get("server_certificate_id").(string)
+		client := meta.(*connectivity.AliyunClient)
+		err := resource.Retry(3*time.Minute, func() *resource.RetryError {
+			raw, err := client.WithSlbClient(func(slbClient *slb.Client) (interface{}, error) {
+				return slbClient.SetDomainExtensionAttribute(req)
+			})
+			if err != nil {
+				if IsExceptedErrors(err, []string{BackendServerConfiguring, DomainExtensionProcessing}) {
+					return resource.RetryableError(WrapErrorf(err, DefaultErrorMsg, "slb_domain_extension", req.GetActionName(), AlibabaCloudSdkGoERROR))
+				}
+				return resource.NonRetryableError(fmt.Errorf("CreateDomainExtension got an error: %#v", err))
+			}
+			addDebug(req.GetActionName(), raw)
+			return nil
+		})
+		if err != nil {
+			return err
+		}
+		d.SetPartial("server_certificate_id")
+	}
+	d.Partial(false)
+	return resourceAliyunSlbDomainExtensionRead(d, meta)
+}
+
+func resourceAliyunSlbDomainExtensionDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	req := slb.CreateDeleteDomainExtensionRequest()
+	req.DomainExtensionId = d.Id()
+	return resource.Retry(5*time.Minute, func() *resource.RetryError {
```

Comment:
Only `DeleteDomainExtension` in this retry, refer to the Delete in the `resource_alicloud_cen_route_entry.go`.